drivers/npmx/npmx_shell.c

```diff
@@ -1792,6 +1792,184 @@ static int cmd_adc_meas_take_vbat(const struct shell *shell, size_t argc, char *
+		}
+		return 0;
+	}
+
+	npmx_timer_t *timer_instance = npmx_timer_get(npmx_instance, 0);
+
+	npmx_timer_config_t timer_config;
+	npmx_error_t err_code = npmx_timer_config_get(timer_instance, &timer_config);
+
+	if (!check_error_code(shell, err_code)) {
+		shell_error(shell, "Error: unable to read timer configuration.");
+		return 0;
+	}
+
+	int err = 0;
+	if (config_type == NPMX_TIMER_CONFIG_TYPE_MODE) {
+		int timer_mode = shell_strtol(argv[1], 0, &err);
+
+		if (err != 0) {
+			shell_error(shell, "Error: timer mode have to be integer.");
+			return 0;
+		}
+
+		switch (timer_mode) {
+		case 0:
+			timer_config.mode = NPMX_TIMER_MODE_BOOT_MONITOR;
+			break;
+		case 1:
+			timer_config.mode = NPMX_TIMER_MODE_WATCHDOG_WARNING;
+			break;
+		case 2:
+			timer_config.mode = NPMX_TIMER_MODE_WATCHDOG_RESET;
+			break;
+		case 3:
+			timer_config.mode = NPMX_TIMER_MODE_GENERAL_PURPOSE;
+			break;
+		case 4:
+			timer_config.mode = NPMX_TIMER_MODE_WAKEUP;
+			break;
+		default:
+			shell_error(shell, "Error: timer mode can be 0-4.");
+			return 0;
+		}
+	} else if (config_type == NPMX_TIMER_CONFIG_TYPE_PRESCALER) {
+		int timer_prescaler = shell_strtol(argv[1], 0, &err);
+
+		if (err != 0) {
+			shell_error(shell, "Error: timer prescaler have to be integer.");
+			return 0;
+		}
+
+		switch (timer_prescaler) {
+		case 0:
+			timer_config.prescaler = NPMX_TIMER_PRESCALER_SLOW;
+			break;
+		case 1:
+			timer_config.prescaler = NPMX_TIMER_PRESCALER_FAST;
+			break;
+		default:
+			shell_error(shell, "Error: timer prescaler value can be 0 or 1.");
+			return 0;
+		}
+	} else if (config_type == NPMX_TIMER_CONFIG_TYPE_COMPARE) {
+		timer_config.compare_value = CLAMP(shell_strtoul(argv[1], 0, &err), 0, 0xFFFFFFUL);
+
+		if (err != 0) {
+			shell_error(shell, "Error: timer period value have to be integer.");
+			return 0;
+		}
+	}
+
+	err_code = npmx_timer_config_set(timer_instance, &timer_config);
+
+	if (check_error_code(shell, err_code)) {
+		if (config_type == NPMX_TIMER_CONFIG_TYPE_MODE) {
+			shell_print(shell, "Success: %d.", timer_config.mode);
+		} else if (config_type == NPMX_TIMER_CONFIG_TYPE_PRESCALER) {
+			shell_print(shell, "Success: %d.", timer_config.prescaler);
+		} else if (config_type == NPMX_TIMER_CONFIG_TYPE_COMPARE) {
+			shell_print(shell, "Success: %d.", timer_config.compare_value);
+		}
+	} else {
+		shell_error(shell, "Error: unable to set timer configuration.");
+	}
+
+	return 0;
+}
+
+static int cmd_timer_config_mode_set(const struct shell *shell, size_t argc, char **argv)
+{
+	return timer_config_set(shell, argc, argv, NPMX_TIMER_CONFIG_TYPE_MODE);
+}
+
+static int cmd_timer_config_prescaler_set(const struct shell *shell, size_t argc, char **argv)
+{
+	return timer_config_set(shell, argc, argv, NPMX_TIMER_CONFIG_TYPE_PRESCALER);
+}
+
+static int cmd_timer_config_period_set(const struct shell *shell, size_t argc, char **argv)
+{
+	return timer_config_set(shell, argc, argv, NPMX_TIMER_CONFIG_TYPE_COMPARE);
```

Comment:
Ordering of the functions' definitions is off, should be:
- timer_config_get
- timer_config_set
- cmd_timer_config_mode_get
- cmd_timer_config_mode_set
- cmd_timer_config_prescaler_get
- cmd_timer_config_prescaler_set
- cmd_timer_config_period_get
- cmd_timer_config_period_set